sqlx-sqlite/src/statement/mod.rs

```diff
@@ -77,3 +78,22 @@ impl ColumnIndex<SqliteStatement<'_>> for &'_ str {
 //         }
 //     }
 // }
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum SqliteOperation {
+    Insert,
+    Update,
+    Delete,
+    Unknown,
+}
+
+impl From<i32> for SqliteOperation {
+    fn from(value: i32) -> Self {
+        match value {
+            SQLITE_INSERT => SqliteOperation::Insert,
+            SQLITE_UPDATE => SqliteOperation::Update,
+            SQLITE_DELETE => SqliteOperation::Delete,
+            _ => SqliteOperation::Unknown,
```

Comment:
I have updated to return `Unknown(i32)`. While not expected, should Sqlite change something under the hood with the return codes this will allow consumers of sqlx to decide how to handle that until sqlx itself is updated.